tests/test_harvest_start_date.py

```diff
@@ -204,88 +78,113 @@ def do_test(self, conn_id):
 
-                # verify that each stream has less records than the first connection sync
+                # Expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+
+                # Collect information for assertions from syncs 1 & 2 base on expected values
+                primary_keys_sync_1 = set([tuple(message.get('data', {}).get(expected_pk) for expected_pk in expected_primary_keys)
+                                           for message in first_sync_records.get(stream, {'messages': []}).get('messages')
+                                           if message.get('action') == 'upsert'])
+                primary_keys_sync_2 = set([tuple(message.get('data', {}).get(expected_pk) for expected_pk in expected_primary_keys)
+                                           for message in second_sync_records.get(stream, {'messages': []}).get('messages')
+                                           if message.get('action') == 'upsert'])
+
+                # Verify that sync 2 has at least one record synced
+                self.assertGreater(second_sync_record_count.get(stream, 0), 0)
+
+                # Verify by primary key values, that all records in the 1st sync are included in the 2nd sync.
+                self.assertTrue(primary_keys_sync_2.issubset(primary_keys_sync_1))
+
+                # Verify that each stream has fewer records than the first connection sync
                 self.assertGreaterEqual(
-                    first_sync_record_count.get(stream, 0),
+                    first_sync_record_count.get(stream, -1),
                     second_sync_record_count.get(stream, 0),
                     msg="second had more records, start_date usage not verified")
 
-                # verify all data from 2nd sync >= start_date
+                # Verify all data from 2nd sync >= start_date
                 target_mark = second_min_bookmarks.get(stream, {"mark": None})
                 target_value = next(iter(target_mark.values()))  # there should be only one
 
                 if target_value:
 
-                    # it's okay if there isn't target data for a stream
+                    # It's okay if there isn't target data for a stream
                     try:
                         target_value = self.local_to_utc(parse(target_value))
 
-                        # verify that the minimum bookmark sent to the target for the second sync
+                        # Verify that the minimum bookmark sent to the target for the second sync
                         # is greater than or equal to the start date
                         self.assertGreaterEqual(target_value,
                                                 self.local_to_utc(parse(state[stream])))
 
                     except (OverflowError, ValueError, TypeError):
-                        print("bookmarks cannot be converted to dates, "
+                        print("Bookmarks cannot be converted to dates, "
```

Comment:
Please change this with tap-tester LOGGER

Reply:
Replaced `print` with LOGGER.